File: COURS/C/C-10/logical_op.c
#include <stdio.h>   // For input/output functions like printf
#include <stdbool.h> // To use 'bool', 'true', and 'false'

// Demonstrating logical operators in C
int main()
{
    int temp = 40;       // Temperature in degrees
    bool isSunny = true; // A boolean indicating whether it's sunny

    // Check if temperature is in the "good" range using logical AND (&&)
    if (temp > 0 && temp < 30)
    {
        // Both conditions must be true for this block to execute
        printf("The temperature is GOOD\n");
    }
    else
    {
        // If either condition is false
        printf("The temperature is BAD\n");
    }

    // Check if it's NOT sunny using logical NOT (!)
    if (!isSunny)
    {
        // This block runs if isSunny is false
        printf("It is CLOUDY outside\n");
    }
    else
    {
        // This block runs if isSunny is true
        printf("It is SUNNY outside\n");
    }

    return 0;
}

File: COURS/JAVA/JAVA-14/LogicalOp.java
public class LogicalOp {
    public static void main(String[] args) {
        // Initialize the temperature and weather condition
        double temp = 20;
        boolean isSunny = true;

        // Check if the temperature is between 0 and 30 inclusive AND it's sunny
        if (temp <= 30 && temp >= 0 && isSunny) {
            System.out.println("The weather is GOOD");
            System.out.println("It is SUNNY outside");
        }
        // Check if the temperature is between 0 and 30 inclusive AND it's not sunny
        else if (temp <= 30 && temp >= 0 && !isSunny) {
            System.out.println("The weather is GOOD");
            System.out.println("It is CLOUDY outside");
        }
        // If the temperature is either above 30 or below 0
        else if (temp > 30 || temp < 0) {
            System.out.println("The weather is BAD");
        }
    }
}
